awx/main/models/jobs.py

```diff
@@ -221,10 +214,8 @@ def network_credential(self):
     def passwords_needed_to_start(self):
         '''Return list of password field names needed to start the job.'''
         needed = []
-        if self.credential:
-            needed.extend(self.credential.passwords_needed)
-        if self.vault_credential:
-            needed.extend(self.vault_credential.passwords_needed)
+        for cred in self.credentials.all():
+            needed.extend(cred.passwords_needed)
```

Comment:
If we want to support prompted passwords for cloud + custom creds in the future, we'll have to rethink our key-based approach here.

Multi-vault support is likely to involve some special case changes here to index necessary vault password by `--vault-id`